### ISSUE TITLE
Transaction version should use the serializer to avoid extra allocations

### ISSUE THREAD

**Issue by cjepson** [decred/dcrd#378]
Currently the transaction version code is very poorly optimized.  It should use the serializer for reading instead of allocating extra memory and copying each time.

**Comment by atriviss** [comment]
Do you mean here: https://github.com/decred/dcrd/blob/master/blockchain/chainio.go#L396

**Comment by davecgh** [comment]
I fixed this back when doing #798.  It was referring to the various version functions such as:

```Go
// DefaultMsgTxVersion returns the default version int32 (serialize the tx
// fully, version number 1).
func DefaultMsgTxVersion() int32 {
	verBytes := make([]byte, 4, 4)
	binary.LittleEndian.PutUint16(verBytes[0:2], TxVersion)
	binary.LittleEndian.PutUint16(verBytes[2:4], uint16(TxSerializeFull))
	ver := binary.LittleEndian.Uint32(verBytes)
	return int32(ver)
}
```

However, I redid the way it works completely in the referenced PR such that to use masks and shifts so it is quite efficient now.
